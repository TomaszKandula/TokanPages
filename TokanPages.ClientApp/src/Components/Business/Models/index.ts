import { ServiceItemDto, TechItemsDto } from "../../../Api/Models";
import { ReactChangeEvent, ReactMouseEvent } from "../../../Shared/types";

export interface MessageFormProps {
    company: string;
    firstName: string;
    lastName: string;
    email: string;
    phone: string;
    description: string;
    techStack: string[];
    services: string[];
}

export interface TechStackListProps {
    isDisabled: boolean;
    list: TechItemsDto[];
    handler: (event: ReactChangeEvent, isChecked: boolean) => void;
}

export interface ServiceItemCardProps {
    isDisabled: boolean;
    key: React.Key | null | undefined;
    value: ServiceItemDto;
    handler: (event: ReactMouseEvent) => void;
    services: string[];
}

export interface BusinessFormProps {
    hasCaption?: boolean;
    hasIcon?: boolean;
    hasShadow?: boolean;
    className?: string;
    background?: string;
}
