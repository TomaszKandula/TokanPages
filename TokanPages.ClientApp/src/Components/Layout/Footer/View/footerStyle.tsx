import { makeStyles } from "@material-ui/core/styles";
import { Colours } from "../../../../Theme";

export const FooterStyle = makeStyles(theme => ({
    page_footer: {
        [theme.breakpoints.down("md")]: {
            textAlign: "center",
        },
        background: Colours.colours.violet,
    },
    icon_box: {
        [theme.breakpoints.down("md")]: {
            width: "100%",
            marginBottom: theme.spacing(0),
        },
        marginLeft: "auto"
    },
    icon: {
        color: Colours.colours.white,
    },
    copyright_box: {
        display: "flex",
        flexWrap: "wrap",
        alignItems: "center",
        paddingTop: 48,
        paddingBottom: 8,
    },
    copyright: {
        fontSize: "1.2rem",
        color: Colours.colours.white,
        [theme.breakpoints.down("md")]: {
            width: "100%",
            order: 12,
        },
    },
    links: {
        color: Colours.colours.white,
        textDecoration: "none",
    },
    version: {
        color: Colours.colours.white,
    },
}));
