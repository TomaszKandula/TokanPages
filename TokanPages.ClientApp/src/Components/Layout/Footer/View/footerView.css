.footer-links {
    color: black;
    text-decoration: none;
}
