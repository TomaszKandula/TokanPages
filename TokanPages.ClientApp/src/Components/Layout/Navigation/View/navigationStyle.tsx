import { makeStyles } from "@material-ui/core/styles";
import { Colours } from "../../../../Theme";

const smallScreenWeb = 1000;
const smallScreenMobile = 600;
const logoHeightSmall = 30;
const logoHeightLarge = 40;

export const NavigationStyle = makeStyles(theme => ({
    app_logo_small: {
        height: logoHeightSmall,
    },
    app_logo_large: {
        height: logoHeightLarge,
    },
    app_bar: {
        background: Colours.colours.white,
    },
    app_left_logo: {
        height: logoHeightSmall,
        alignSelf: "center",
    },
    app_full_logo: {
        [theme.breakpoints.down(smallScreenMobile)]: {
            display: "none",
        },
        height: logoHeightSmall,
        marginLeft: "auto",
        marginRight: "auto",
        alignSelf: "center",
    },
    app_just_logo: {
        [theme.breakpoints.down(smallScreenMobile)]: {
            display: "block",
        },
        [theme.breakpoints.up(smallScreenMobile)]: {
            display: "none",
        },
        height: logoHeightLarge,
        marginLeft: "auto",
        marginRight: "auto",
        alignSelf: "center",
    },
    app_link: {
        [theme.breakpoints.down(smallScreenMobile)]: {
            justifyContent: "right",
        },
        display: "flex",
        justifyContent: "center",
    },
    tool_bar: {
        display: "flex",
        flexDirection: "row",
        flexWrap: "nowrap",
        justifyContent: "space-between",
    },
    nav_large_screen: {
        [theme.breakpoints.down(smallScreenWeb)]: {
            display: "none",
        },
        [theme.breakpoints.up(smallScreenWeb)]: {
            display: "block",
        },
    },
    nav_small_screen: {
        [theme.breakpoints.down(smallScreenWeb)]: {
            display: "block",
        },
        [theme.breakpoints.up(smallScreenWeb)]: {
            display: "none",
        },
    },
    nav_menu: {
        color: Colours.colours.black,
        display: "flex",
    },
    nav_icon: {
        height: 48,
        marginTop: 6,
        marginBottom: 6,
    },
    nav_items: {
        display: "flex",
        alignItems: "center",
    },
    nav_left: {
        justifyContent: "flex-start",
    },
    nav_right: {
        justifyContent: "flex-end",
    },
    nav_centre: {
        justifyContent: "center",
    },
    languages_wrapper: {
        display: "flex",
        alignItems: "center",
    },
    languagesBox: {
        [theme.breakpoints.down(smallScreenMobile)]: {
            marginRight: 0,
        },
        marginRight: 30,
        alignSelf: "center",
    },
    languages_control: {},
    languages_selection: {
        color: Colours.colours.black,
        width: 75,
    },
    languages_check: {
        paddingLeft: 5,
        color: Colours.colours.darkViolet1,
    },
    user_avatar: {
        [theme.breakpoints.down(smallScreenWeb)]: {
            display: "none",
        },
        display: "flex",
    },
    flag_image: {
        height: 14,
        width: 14,
        marginRight: 5,
    },
}));
