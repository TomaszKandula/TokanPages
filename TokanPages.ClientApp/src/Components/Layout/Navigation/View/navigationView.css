.navigation-app-logo-small {
    height: 30px !important;
}

.navigation-app-logo-large {
    height: 40px !important;
}

.navigation-app-left-logo {
    height: 30px !important;
    width: 181px !important;
    align-self: center !important;
}

.navigation-tool-bar {
    padding-left: 24px;
    padding-right: 24px;
    display: flex;
    flex-direction: row;
    flex-wrap: nowrap;
    justify-content: space-between;
    align-items: center;
}

.navigation-nav-menu {
    color: black;
    display: flex;
}

.navigation-nav-icon {
    height: 48px;
    margin-top: 6px;
    margin-bottom: 6px;
}

.navigation-nav-items {
    display: flex;
    align-items: center;
}

.navigation-nav-left {
    justify-content: flex-start;
}

.navigation-nav-right {
    justify-content: flex-end;
}

.navigation-nav-centre {
    justify-content: center;
}

.navigation-nav-back {
    margin-top: 8px;
    margin-bottom: 8px;
    margin-left: 8px;
}

.navigation-languages-check {
    padding-left: 5px;
    color: #4649c3;
}

.navigation-flag-image {
    height: 14px;
    width: 14px;
    margin-right: 5px;
}

.navigation-app-full-logo {
    display: block;
    height: 30px;
    margin-left: auto;
    margin-right: auto;
    align-self: center;
}

@media (max-width: 768px) {
    .navigation-app-full-logo {
        display: none;
    }
}

.navigation-app-just-logo {
    display: none;
    height: 40px;
    margin-left: auto;
    margin-right: auto;
    align-self: center;
}

@media (max-width: 768px) {
    .navigation-app-just-logo {
        display: block;
    }
}

.navigation-user-avatar {
    display: none;
}

@media (min-width: 1024px) {
    .navigation-user-avatar {
        display: flex;
    }
}

.navigation-languages-box {
    margin-right: 30px;
    align-self: center;
}

@media (max-width: 1024px) {
    .navigation-languages-box {
        margin-right: 0;
    }
}

.navigation-nav-large-screen {
    display: block;
}

@media (max-width: 1024px) {
    .navigation-nav-large-screen {
        display: none;
    }
}

.navigation-nav-small-screen {
    display: none;
}

@media (max-width: 1024px) {
    .navigation-nav-small-screen {
        display: block;
    }
}

.navigation-nav-drawer-root {
    top: 0;
    flex: 1 0 auto;
    height: 100%;
    display: flex;
    outline: 0;
    z-index: 1200;
    position: fixed;
    overflow-y: auto;
    flex-direction: column;
    transition: left 350ms cubic-bezier(0, 0, 0.2, 1);
}

.navigation-nav-drawer-backdrop {
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    display: flex;
    z-index: 1000;
    position: fixed;
    align-items: center;
    justify-content: center;
    background-color: rgba(0, 0, 0, 0.5);
    -webkit-tap-highlight-color: transparent;
    transition: opacity 350ms cubic-bezier(0.4, 0, 0.2, 1);
}

.navigation-nav-drawer-container {
    z-index: 1100;
    height: 100%;
    width: 300px;
    background-color: white;
}

.navigation-nav-drawer-hero {
    height: 200px;
    display: flex;
    justify-content: center;
    align-items: center;
    background-color: #ebecfb;
}

.navigation-nav-drawer-logo {
    height: 120px;
    margin-left: auto;
    margin-right: auto;
    align-self: center;
}
