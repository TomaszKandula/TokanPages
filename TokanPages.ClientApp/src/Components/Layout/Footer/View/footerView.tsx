import * as React from "react";
import { IconDto, LinkDto } from "../../../../Api/Models";
import { Icon, IconButton, Link, ProgressBar } from "../../../../Shared/Components";
import { v4 as uuidv4 } from "uuid";
import "./footerView.css";

interface LegalInfoProps {
    copyright: string;
    reserved: string;
}

interface Properties {
    isLoading: boolean;
    terms: LinkDto;
    policy: LinkDto;
    versionInfo: string;
    hasVersionInfo: boolean;
    legalInfo: LegalInfoProps;
    hasLegalInfo: boolean;
    icons: IconDto[];
    hasIcons: boolean;
    className?: string;
}

const SetTermsLink = (props: Properties): React.ReactElement => (
    <Link to={props?.terms?.href ?? ""} className="footer-links ml-2 mr-2">
        <span>{props?.terms?.text}</span>
    </Link>
);

const SetPolicyLink = (props: Properties): React.ReactElement => (
    <Link to={props?.policy?.href ?? ""} className="footer-links ml-2 mr-2">
        <span>{props?.policy?.text}</span>
    </Link>
);

export const FooterView = (props: Properties): React.ReactElement => {
    return (
        <footer className={`bulma-footer ${props.className ?? ""}`}>
            {props.isLoading 
            ? <ProgressBar size={32} /> 
            : <div className="bulma-content has-text-centered">
                <p className="is-size-5">
                    <span className="ml-2 mr-2">
                        {props?.legalInfo.copyright}
                    </span>
                    <span>|</span>
                    <span className="ml-2 mr-2">
                        {props?.legalInfo.reserved}
                    </span>
                    <span>|</span>
                    <SetTermsLink {...props} />
                    <span>|</span>
                    <SetPolicyLink {...props} />
                </p>
                <p className="is-size-6">
                    {props?.versionInfo}
                </p>
                <div className="is-flex is-justify-content-center">
                {props?.icons?.map((item: IconDto, _index: number) => (
                    <Link 
                        to={item.href}
                        key={uuidv4()}
                        aria-label={item.name}
                    >
                        <IconButton>
                            <Icon name={item.name} size={1.5} />
                        </IconButton>
                    </Link>
                ))}
                </div>
            </div>}
        </footer>
    );
};
