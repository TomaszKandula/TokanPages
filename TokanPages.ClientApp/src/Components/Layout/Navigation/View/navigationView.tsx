import * as React from "react";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import IconButton from "@material-ui/core/IconButton";
import Avatar from "@material-ui/core/Avatar";
import MenuIcon from "@material-ui/icons/Menu";
import { FormControl, Grid, MenuItem, Select, Typography, Box } from "@material-ui/core";
import { ILanguageItem } from "../../../../Api/Models/";
import { IApplicationLanguage } from "../../../../Store/States/";
import { HideOnScroll } from "../../../../Shared/Components/Scroll";
import { IItem } from "../../../../Shared/Components/ListRender/Models";
import { GET_AVATARS_URL } from "../../../../Api/Request";
import { SideMenuView } from "./../SideMenu/sideMenuView";
import { NavigationStyle } from "./navigationStyle";
import { v4 as uuidv4 } from "uuid";
import Validate from "validate.js";

interface IBinding
{
    bind: IProperties;
}

interface IProperties
{
    isLoading: boolean;
    drawerState: { open: boolean };
    openHandler: any;
    closeHandler: any;
    infoHandler: any;
    isAnonymous: boolean;
    logo: string;
    avatarName: string;
    userAliasText: string;
    languages: IApplicationLanguage;
    languageId: string;
    languageHandler: any;
    menu: { image: string, items: IItem[] };
}

interface IRenderLanguageSelection extends IBinding
{
    styleControl?: string; 
    styleSelect?: string; 
    styleMenu?: string;
}

const RenderMenuIcon = (props: IBinding): JSX.Element => 
{
    const classes = NavigationStyle();
    return(
        <IconButton color="inherit" aria-label="menu" 
            onClick={props.bind?.openHandler} className={classes.nav_icon}>
            <MenuIcon />
        </IconButton>
    );
}

const RenderAvatar = (props: IBinding): JSX.Element => 
{
    if (props.bind?.isAnonymous)
    {
        return(<Avatar>A</Avatar>);
    }

    if (Validate.isEmpty(props.bind?.avatarName))
    {
        const userLetter = props.bind?.userAliasText?.charAt(0).toUpperCase();
        return(<Avatar>{userLetter}</Avatar>);
    }

    const avatarName = `${GET_AVATARS_URL}/${props.bind?.avatarName}`;
    return(<Avatar alt="Avatar" src={avatarName} />);
}

const RenderLanguageSelection = (props: IRenderLanguageSelection): JSX.Element => 
{
    return(
        <FormControl className={props.styleControl}>
            <Select value={props.bind?.languageId} onChange={props.bind?.languageHandler} disableUnderline className={props.styleSelect}>
                {props.bind?.languages?.languages.map((item: ILanguageItem, _index: number) => (
                    <MenuItem value={item.id} key={uuidv4()} className={props.styleMenu}>
                        {item.name}
                    </MenuItem>
                ))}
            </Select>
        </FormControl>
    );
}

const RenderContent = (props: IBinding): JSX.Element => 
{
    const classes = NavigationStyle();
    return(
        <>
            <Box className={classes.languagesBox}>
                <RenderLanguageSelection 
                    {...props} 
                    styleSelect={classes.languages_selection} 
                    styleMenu={classes.languages_menu}
                />
            </Box>
            <div className={classes.user_avatar}>
                <IconButton color="inherit" onClick={props.bind?.infoHandler} >
                    <RenderAvatar {...props} /> 
                </IconButton>
            </div>
        </>
    );
}

export const NavigationView = (props: IBinding): JSX.Element => 
{
    const classes = NavigationStyle();
    const fullName = "</> tom kandula";
    const justLogo = "</>";
    return (
        <HideOnScroll {...props}>
            <AppBar className={classes.app_bar}>
                <Toolbar className={classes.tool_bar}>

                    <Grid container item xs={12} spacing={3}>
                        <Grid item xs className={classes.nav_menu}>
                            {props.bind?.isLoading ? null : <RenderMenuIcon {...props} />}
                        </Grid>
                        <Grid item xs className={classes.app_link}>
                            <Typography className={classes.app_full_logo}>{fullName}</Typography>
                            <Typography className={classes.app_just_logo}>{justLogo}</Typography>
                        </Grid>
                        <Grid item xs className={classes.content_right_side}>
                            {props.bind?.isLoading ? null : <RenderContent {...props} />}
                        </Grid>
                    </Grid>

                </Toolbar>

                <SideMenuView bind=
                {{
                    drawerState: props.bind.drawerState,
                    closeHandler: props.bind.closeHandler,
                    isAnonymous: props.bind.isAnonymous,
                    menu: props.bind.menu
                }}/>

            </AppBar>
        </HideOnScroll>
    );
}
