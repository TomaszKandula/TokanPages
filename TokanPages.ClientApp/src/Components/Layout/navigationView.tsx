import * as React from "react";
import { Link } from "react-router-dom";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import IconButton from '@material-ui/core/IconButton';
import Avatar from '@material-ui/core/Avatar';
import MenuIcon from '@material-ui/icons/Menu';
import { Grid, Typography } from "@material-ui/core";
import HideOnScroll from "../../Shared/Components/Scroll/hideOnScroll";
import { ICONS_PATH } from "../../Shared/constants";
import { renderImage } from "../../Shared/Components/CustomImage/customImage";
import { IItem } from "../../Shared/Components/ListRender/Models/item";
import MenuView from "./Components/menuView";
import navigationStyle from "./Styles/navigationStyle";

interface IBinding
{
    bind: IProperties;
}

interface IProperties
{
    drawerState: { open: boolean };
    openHandler: any;
    closeHandler: any;
    isAnonymous: boolean;
    logo: string;
    avatar: string;
    anonymousText: string;
    userAliasText: string;
    menu: { image: string, items: IItem[] };
}

const NavigationView = (props: IBinding): JSX.Element => 
{
    const classes = navigationStyle();
    return (
        <HideOnScroll {...props}>
            <AppBar className={classes.appBar}>
                <Toolbar className={classes.toolBar}>

                    <Grid container item xs={12} spacing={3}>
                        <Grid item xs className={classes.menu}>
                            <IconButton color="inherit" aria-label="menu" onClick={props.bind.openHandler}>
                                <MenuIcon />
                            </IconButton>
                        </Grid>
                        <Grid item xs className={classes.link}>
                            <Link to="/">
                                <div data-aos="fade-down" className={classes.image} >
                                    {renderImage(ICONS_PATH, props.bind.logo, classes.logo)}
                                </div>
                            </Link>
                        </Grid>
                        <Grid item xs className={classes.avatar}>
                            <Typography className={classes.userAlias} component={"span"} variant={"body1"} >
                                {props.bind.isAnonymous ? props.bind.anonymousText : props.bind.userAliasText}
                            </Typography>
                            <IconButton color="inherit">
                                <Avatar alt="Avatar" src={props.bind.avatar} /> 
                            </IconButton>
                        </Grid>
                    </Grid>

                </Toolbar>

                <MenuView bind=
                {{
                    drawerState: props.bind.drawerState,
                    closeHandler: props.bind.closeHandler,
                    isAnonymous: props.bind.isAnonymous,
                    menu: props.bind.menu
                }}/>

            </AppBar>
        </HideOnScroll>
    );
}

export default NavigationView;
