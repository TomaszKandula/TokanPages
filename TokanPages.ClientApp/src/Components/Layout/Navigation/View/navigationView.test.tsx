import "../../../../setupTests";
import React from "react";
import { shallow } from "enzyme";
import { NavigationView } from "../../Navigation/View/navigationView";
import { IItem } from "../../../../Shared/Components/ListRender/Models";
import { IUserLanguage } from "../../../../Store/States";

describe("Test component: featuresView.", () => 
{
    it("Renders correctly '<NavigationView />' when content is loaded.", () => 
    {
        const items: IItem = 
        {
            id: "79a6c65d-08b8-479b-9507-97feb05e30c2",
            type: "item",
            value: "Home",
            link: "/",
            icon: "Home",
            enabled: true
        };

        const languages: IUserLanguage = 
        {
            id: "eng",
            languages: 
            [
                {
                    id: "eng", 
                    name: "English", 
                    isDefault: true
                },
                {
                    id: "pol", 
                    name: "Polski", 
                    isDefault: false
                }
            ]
        }

        const tree = shallow(<NavigationView bind=
        {{
            isLoading: false,
            drawerState: 
            { 
                open: false 
            },
            openHandler: jest.fn(),
            closeHandler: jest.fn(),
            infoHandler: jest.fn(),
            isAnonymous: false,
            logo: "",
            avatarName: "",
            userAliasText: "",
            menu: 
            { 
                image: "", 
                items: [items] 
            },
            languages: languages,
            languageId: "eng",
            languageHandler: jest.fn(),
        }}/>);
        expect(tree).toMatchSnapshot();
    });
});
