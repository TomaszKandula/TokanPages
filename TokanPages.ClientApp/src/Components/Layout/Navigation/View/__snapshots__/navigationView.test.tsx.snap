// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component: featuresView should render correctly '<NavigationView />' when content is loaded. 1`] = `
<header
  class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-app_bar-3 mui-fixed MuiPaper-elevation0"
>
  <div
    class="makeStyles-nav_large_screen-9"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular makeStyles-tool_bar-31 MuiToolbar-gutters"
    >
      <div
        class="MuiBox-root MuiBox-root-47 makeStyles-nav_menu-34 makeStyles-nav_left-37"
      >
        <a
          class="makeStyles-app_logo_small-24"
          href="/"
        >
          <img
            alt=""
            class="makeStyles-app_left_logo-27"
            src="http://localhost:5000/api/v1/content/assets/getNonVideoAsset?blobName=images/icons/logo.svg"
          />
        </a>
      </div>
      <div
        class="MuiBox-root MuiBox-root-48 makeStyles-nav_items-36 makeStyles-nav_centre-39"
      >
        <ul
          class="MuiList-root makeStyles-list-49 MuiList-padding"
        />
      </div>
      <div
        class="MuiBox-root MuiBox-root-50 makeStyles-nav_items-36 makeStyles-nav_right-38"
      >
        <div
          class="MuiBox-root MuiBox-root-74 makeStyles-languagesBox-68"
        >
          <div
            class="MuiFormControl-root makeStyles-languages_control-69"
          >
            <div
              class="MuiInputBase-root MuiInput-root makeStyles-languages_selection-70 MuiInputBase-formControl MuiInput-formControl"
            >
              <div
                aria-haspopup="listbox"
                class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                role="button"
                tabindex="0"
              >
                <div
                  class="makeStyles-languages_wrapper-91"
                >
                  <img
                    alt=""
                    class="makeStyles-flag_image-97"
                    src="http://localhost:5000/api/v1/content/assets/getNonVideoAsset?blobName=images/flags/eng.png"
                  />
                  <div>
                    ENG
                  </div>
                </div>
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput"
                tabindex="-1"
                value="eng"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSelect-icon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="makeStyles-user_avatar-119"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault"
              />
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    class="makeStyles-nav_small_screen-10"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular makeStyles-tool_bar-128 MuiToolbar-gutters"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          class="MuiGrid-root makeStyles-nav_menu-131 makeStyles-nav_left-134 MuiGrid-item MuiGrid-grid-xs-true"
        >
          <button
            aria-label="menu"
            class="MuiButtonBase-root MuiIconButton-root makeStyles-nav_icon-155 MuiIconButton-colorInherit"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                />
              </svg>
            </span>
          </button>
        </div>
        <div
          class="MuiGrid-root makeStyles-nav_items-133 makeStyles-nav_centre-136 MuiGrid-item MuiGrid-grid-xs-true"
        >
          <a
            class="makeStyles-app_logo_small-121"
            href="/"
          >
            <img
              alt=""
              class="makeStyles-app_full_logo-125"
              src="http://localhost:5000/api/v1/content/assets/getNonVideoAsset?blobName=images/icons/logo.svg"
            />
          </a>
          <a
            class="makeStyles-app_logo_large-122"
            href="/"
          />
        </div>
        <div
          class="MuiGrid-root makeStyles-nav_items-133 makeStyles-nav_right-135 MuiGrid-item MuiGrid-grid-xs-true"
        >
          <div
            class="MuiBox-root MuiBox-root-190 makeStyles-languagesBox-184"
          >
            <div
              class="MuiFormControl-root makeStyles-languages_control-185"
            >
              <div
                class="MuiInputBase-root MuiInput-root makeStyles-languages_selection-186 MuiInputBase-formControl MuiInput-formControl"
              >
                <div
                  aria-haspopup="listbox"
                  class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="makeStyles-languages_wrapper-207"
                  >
                    <img
                      alt=""
                      class="makeStyles-flag_image-213"
                      src="http://localhost:5000/api/v1/content/assets/getNonVideoAsset?blobName=images/flags/eng.png"
                    />
                    <div>
                      ENG
                    </div>
                  </div>
                </div>
                <input
                  aria-hidden="true"
                  class="MuiSelect-nativeInput"
                  tabindex="-1"
                  value="eng"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSelect-icon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="makeStyles-user_avatar-235"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <div
                  class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault"
                />
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</header>
`;
